### PR TITLE
unit: stop journald before SIGKILL on soft-reboot

### PR DESCRIPTION
**Pull request by yuwata**
workaround for #30195.

### CODE REVIEW

File: units/systemd-journald.service.in [hunk 1]
```diff
@@ -15,6 +15,11 @@ Requires=systemd-journald.socket
 After=systemd-journald.socket systemd-journald-dev-log.socket systemd-journald-audit.socket syslog.socket
 Before=sysinit.target
 
+# To avoid journald SIGKILLed during soft-reboot and corrupting journals.
+# See https://github.com/systemd/systemd/issues/30195
+Before=soft-reboot.target
```
Comment (dtardon) [inline units/systemd-journald.service.in:20]: Shouldn't this be `shutdown.target` instead, to cover regular reboot/shutdown too?

Comment (dtardon) [reply units/systemd-journald.service.in:20]: Also, `systemd-journald@.service.in` likely needs this too.

Comment (yuwata) [reply units/systemd-journald.service.in:20]: No, that's intended. On shutdown or regular reboot, systemd-journald will be killed by systemd-shutdown on SIGTERM.
systemd-shutdown wait for processes being terminated, so journald can safely sync and close journal files.

When shutdown.target is used here, then we cannot log anything about shutting down process.

Comment (yuwata) [reply units/systemd-journald.service.in:20]: > Also, `systemd-journald@.service.in` likely needs this too.

Indeed. Thanks.

**Review by YHNdnzj — APPROVED** [review]

**Comment by poettering** [comment]
lgtm

**Comment by bluca** [comment]
@mrc0mmand could you please restart the centos job? looks like a network issue

**Comment by mrc0mmand** [comment]
> @mrc0mmand could you please restart the centos job? looks like a network issue

Hmpf, there's something fishy going on, the machine takes noticeably longer to boot and sometimes it fails to come up completely. But it doesn't seem to be caused by this PR as I see it in other PRs as well, will need to look into that.

**Comment by mrc0mmand** [comment]
Oh well:

```
[    5.786022] BUG: unable to handle kernel NULL pointer dereference at 00000000000000b8
[    5.787033] PGD 0 P4D 0 
[    5.787356] Oops: 0000 [#1] SMP NOPTI
[    5.787821] CPU: 3 PID: 672 Comm: kworker/3:3 Not tainted 4.18.0-526.el8.x86_64 #1
[    5.788761] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.16.2-1.fc38 04/01/2014
[    5.789852] Workqueue: cgwb_release cgwb_release_workfn
[    5.790509] RIP: 0010:cgwb_release_workfn+0xf2/0x180
[    5.791132] Code: 2d 8d 48 89 43 50 48 05 00 01 00 00 48 89 43 58 c6 07 00 0f 1f 40 00 fb 66 0f 1f 44 00 00 4c 89 e7 e8 72 fe ff ff 48 8b 43 48 <48> 8b 90 b8 00 00 00 48 05 b8 00 00 00 48 39 d0 75 40 48 89 df 48
[    5.793440] RSP: 0018:ffffb6a2c079be78 EFLAGS: 00010282
[    5.794092] RAX: 0000000000000000 RBX: ffff8e8f43a962f8 RCX: 000000008010000e
[    5.794976] RDX: 000000008010000f RSI: 000000008010000e RDI: ffff8e8f41002fc0
[    5.795872] RBP: ffffffff8d36e9c0 R08: 0000000000000001 R09: 0000000000000001
[    5.796765] R10: ffff8e8f45cdd400 R11: 0000000000003600 R12: ffff8e8f43a96000
[    5.797661] R13: ffff8e8f4562c480 R14: 0000000000000000 R15: ffff8e8f43a96300
[    5.798553] FS:  0000000000000000(0000) GS:ffff8e8fbfd80000(0000) knlGS:0000000000000000
[    5.799570] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    5.800291] CR2: 00000000000000b8 CR3: 000000000547c000 CR4: 0000000000350ee0
[    5.801190] Call Trace:
[    5.801515]  ? __die_body+0x1a/0x60
[    5.801969]  ? no_context+0x1ba/0x3f0
[    5.802439]  ? srso_return_thunk+0x5/0x5f
[    5.802951]  ? __bad_area_nosemaphore+0x16c/0x1c0
[    5.803560]  ? do_page_fault+0x37/0x12d
[    5.804046]  ? page_fault+0x1e/0x30
[    5.804492]  ? cgwb_release_workfn+0xf2/0x180
[    5.805047]  ? cgwb_release_workfn+0xee/0x180
[    5.805601]  process_one_work+0x1d3/0x390
[    5.806118]  ? process_one_work+0x390/0x390
[    5.806661]  worker_thread+0x30/0x390
[    5.807124]  ? process_one_work+0x390/0x390
[    5.807664]  kthread+0x134/0x150
[    5.808081]  ? set_kthread_struct+0x50/0x50
[    5.809894]  ret_from_fork+0x35/0x40
[    5.810381] Modules linked in: intel_rapl_msr intel_rapl_common amd_energy kvm_amd ccp kvm irqbypass crct10dif_pclmul crc32_pclmul ghash_clmulni_intel pcspkr virtio_balloon i2c_piix4 joydev loop fuse nfnetlink xfs libcrc32c ata_generic cirrus drm_shmem_helper drm_kms_helper syscopyarea sysfillrect sysimgblt ata_piix drm libata crc32c_intel virtio_net net_failover failover virtio_blk serio_raw dm_mirror dm_region_hash dm_log dm_mod
[    5.815164] Red Hat flags: eBPF/cgroup
[    5.815648] CR2: 00000000000000b8
[    5.816072] ---[ end trace f26f9613591c1052 ]---
[    5.816658] RIP: 0010:cgwb_release_workfn+0xf2/0x180
[    5.817283] Code: 2d 8d 48 89 43 50 48 05 00 01 00 00 48 89 43 58 c6 07 00 0f 1f 40 00 fb 66 0f 1f 44 00 00 4c 89 e7 e8 72 fe ff ff 48 8b 43 48 <48> 8b 90 b8 00 00 00 48 05 b8 00 00 00 48 39 d0 75 40 48 89 df 48
[    5.819642] RSP: 0018:ffffb6a2c079be78 EFLAGS: 00010282
[    5.820296] RAX: 0000000000000000 RBX: ffff8e8f43a962f8 RCX: 000000008010000e
[    5.821187] RDX: 000000008010000f RSI: 000000008010000e RDI: ffff8e8f41002fc0
[    5.822141] RBP: ffffffff8d36e9c0 R08: 0000000000000001 R09: 0000000000000001
[    5.823042] R10: ffff8e8f45cdd400 R11: 0000000000003600 R12: ffff8e8f43a96000
[    5.823936] R13: ffff8e8f4562c480 R14: 0000000000000000 R15: ffff8e8f43a96300
[    5.824834] FS:  0000000000000000(0000) GS:ffff8e8fbfd80000(0000) knlGS:0000000000000000
[    5.825850] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    5.826584] CR2: 00000000000000b8 CR3: 000000000547c000 CR4: 0000000000350ee0
[    5.827480] Kernel panic - not syncing: Fatal exception
[    5.828243] Kernel Offset: 0x9a00000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
[    5.829599] ---[ end Kernel panic - not syncing: Fatal exception ]---
```

The good thing is that we didn't screw anything up. I'll prep a workaround for the C8S job, but in the meantime please ignore the results.

**Comment by bluca** [comment]
It also happens in the mkosi centos8 job, same kernel panic: https://github.com/systemd/systemd/actions/runs/7016725652/job/19088525951?pr=30220

**Comment by bluca** [comment]
Can we do the same in mkosi?

**Comment by mrc0mmand** [comment]
I guess so, just install `kernel-4.18.0-521.el8` (latest working kernel) instead of just `kernel`.

**Pull request merged**